usr/src/uts/common/fs/zfs/sys/zil_impl.h

```diff
@@ -142,6 +141,8 @@ typedef struct zil_bp_node {
 
 #define	ZIL_MAX_LOG_DATA (SPA_OLD_MAXBLOCKSIZE - sizeof (zil_chain_t) - \
     sizeof (lr_write_t))
+#define	ZIL_MAX_COPIED_DATA \
+    ((SPA_OLD_MAXBLOCKSIZE - sizeof (zil_chain_t)) / 2 - sizeof (lr_write_t))
 
```

Comment:
This maximum value is based on assumption that we should fit at least two log records into single maximum-sized log block.  Such block has maximal size of SPA_OLD_MAXBLOCKSIZE, and includes zil_chain_t header per block, and lr_write_t header per (each of two) record.  If data block is bigger then this value, then we should not assume we'll be lucky to fit it, and should postpone the copying till later stage, when it can be done more efficiently.